hooks/helpers/useContractInteractionNotifications.ts

```diff
@@ -0,0 +1,42 @@
+import { useNotificationsContext } from "hooks";
+import { ContractInteractionNotificationT } from "types";
+
+export function useContractInteractionNotifications() {
```

Comment:
This hook handles notifications for contract transactions and receipts. transactions are by definition pending, and receipts are settled.